src/gc.c

```diff
@@ -2242,7 +2242,7 @@ JL_DLLEXPORT int jl_gc_mark_queue_obj(jl_ptls_t ptls, jl_value_t *obj)
 JL_DLLEXPORT void jl_gc_mark_queue_objarray(jl_ptls_t ptls, jl_value_t *parent,
                                             jl_value_t **objs, size_t nobjs)
 {
-    uintptr_t nptr = (nobjs << 2) & (jl_astaggedvalue(parent)->bits.gc & 3);
+    uintptr_t nptr = (nobjs << 2) | (jl_astaggedvalue(parent)->bits.gc & 2);
```

Comment:
Thank you for the fix!

The previous code always would end up with `nptr` equal to zero, so clearly was wrong. It was added just recently, replacing code that was in place a couple years, and that old code used effectively `bits.gc & 2`, so that change also seems plausible to me -- albeit without having looked further into `gc_mark_objarray`. But I have no reason to mistrust your judgement on this, just pointing out the limits to which I understand/approve this :-).